issue 71: reading from java beans

kuba-mikulski opened the pull request.

krzemin commented:
Also, please remember about running scalafmt before comitting - otherwise travis build will fail.

krzemin commented on chimney/src/main/scala/io/scalaland/chimney/internal/MacroUtils.scala:
```diff
@@ -26,9 +26,9 @@ trait MacroUtils extends CompanionUtils {
+    def parameterlessMethods: Iterable[MethodSymbol] = {
       t.decls.collect {
-        case m: MethodSymbol if m.isGetter =>
+        case m: MethodSymbol if (m.paramLists.isEmpty || m.paramLists == List(List())) && !m.isPrivate =>
```
Note that this is also changed in #76 in order for one test to pass. Whichever PR comes first, please keep their behavior (can be factored out to a separate method of course).

krzemin commented:
Travis build failed due to code being not properly formatted. Please run `scalafmt` sbt task and push again.

kuba-mikulski commented:
Addressed the comments in this PR, but Codacy is now complaining due to too many ifs in `DslBlackboxMacros`, and I don't really see a way of converting it to case matching due to the values being dynamic. Thoughts?

krzemin commented:
I think you can ignore Codacy - it was complaining for every such if-branch. Instead, I would focus on adding few more test cases to keep the coverage. At least:
- case when the switch is disabled and we don't interpret bean naming conventions
- case when a field named `isXXX` doesn't have Boolean type
